src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
Well, but that's my point, I understand it's an implementation detail, but that details is currently spread through 3 different projects: 
- `IMemoryStorageGrain`: defined in `Orleans`
- `MemoryStorageGrain`: defined in `OrleansRuntime`
- `MemoryStorage`: defined in `OrleansProviders`

Anyway, agree on not inheriting from OrleansException, and will make this internal (and add InternalsVisiblesTo) and move to a storage specific namespace